modules/benchmark/R/align_data_to_data_pft.R

```diff
@@ -0,0 +1,196 @@
+      
+      aligned_species_list$final<-bety_codes_intersect
+      
+      return(aligned_species_list) #Returns the intersection of the speceies lists
+    }else{
+      PEcAn.logger::logger.warn("These observations cannot be aligned, as they have different species lists. Returning NULL. Check species lists, or  set 'subset_are_ok' to TRUE. ")
+      return(NULL)
+    }
+    
+  }else if(check_if_species_list(format_one) && !check_if_species_list(format_two)){
+    
+    if(is.null(custom_table)){logger.severe("Please provide custom_table")}else if (!is.null(custom_table))
+    {
+      if(check_if_legal_table(custom_table, observation_one, observation_two)){
+        
+        translation_table<-NULL
+        if (get_species_list_standard(format_one)=="custom"){tanslation_table<-custom_table}
+        
+        bety_codes_one<-PEcAn.data.land::match_species_id(input_codes=observation_one, format_name= get_species_list_standard(format_one),translation_table = translation_table, bety=con)
+        
+        aligned_by_one<-align_by_observation_one(observation_one,observation_two, custom_table)
+        aligned_by_two<-align_by_observation_two(observation_one,observation_two, custom_table)
+        
+        aligned_species_list<-list()
+        aligned_species_list$bety_species_id$observation_one<-bety_codes_one
+        aligned_species_list$bety_species_id$observation_two<-NA
+        aligned_species_list$original$observation_one<-observation_one
+        aligned_species_list$original$observation_two<-observation_two
+        aligned_species_list$aligned$aligned_by_observation_one<-aligned_by_one
+        aligned_species_list$aligned$aligned_by_observation_two<-aligned_by_two
+        
+        return(aligned_species_list)
+        
+        
+      }else{
+        logger.severe("custom_table provided does not correctly map plant_function_type_one to plant_functional_type_two. One or more rows are mapped to multiple plant funcitonal types.")
+      } 
+    }
+   
+  
+    
+  }else if(!check_if_species_list(format_one) && check_if_species_list(format_two)){
+    
+    if(is.null(custom_table)){PEcAn.logger::logger.severe("Please provide custom_table")}else if (!is.null(custom_table))
+    {
+      if(check_if_legal_table(custom_table, observation_one, observation_two)){
+        
+        translation_table<-NULL
+        if (get_species_list_standard(format_two)=="custom"){tanslation_table<-custom_table}
+  
+        bety_codes_two<-PEcAn.data.land::match_species_id(input_codes=observation_two, format_name= get_species_list_standard(format_two),translation_table = translation_table, bety=con)
+        
+        aligned_by_one<-align_by_observation_one(observation_one,observation_two, custom_table)
+        aligned_by_two<-align_by_observation_two(observation_one,observation_two, custom_table)
+        
+        aligned_species_list<-list()
+        aligned_species_list$bety_species_id$observation_one<-bety_codes_one
```

Comment:
Is this supposed to be bety_codes_one or bety_codes_two? If one, then why calc two above?